### ISSUE TITLE
Swipe to Add to Collection

### ISSUE THREAD

**Issue by sw0rdfish** [Dimillian/ACHNBrowserUI#112]
Hey all. New here. Gonna try to jump in with helping on things but I’ve never coded in swift. See if I can catchup fast. 

Would be better if swiping right added a bug/fish/item to your collection. I also find a check might be better than a star for the non swipe method? It took me a bit to figure out star wasn’t favouriting meaning something else. Thoughts?

**Comment by Dimillian** [comment]
<img width="559" alt="Screenshot 2020-05-11 at 19 41 58" src="https://user-images.githubusercontent.com/535509/81593304-8a817d00-93bf-11ea-979f-9b732e406afe.png">
<img width="559" alt="Screenshot 2020-05-11 at 19 42 11" src="https://user-images.githubusercontent.com/535509/81593319-8e150400-93bf-11ea-9ffc-8d5df9a6f9e3.png">
<img width="559" alt="Screenshot 2020-05-11 at 19 42 17" src="https://user-images.githubusercontent.com/535509/81593323-8f463100-93bf-11ea-89de-8b6bb2c0350c.png">
So it's a heart for villager, a checkmark for critters and a star for items. But we could *also* support swipe. It's a bit less discoverable that why I think the one tap to add from the row is also good.

**Comment by sw0rdfish** [comment]
Weird. Mine has stars for the critters. Is that a bug ? Maybe I’m behind a version

**Comment by sw0rdfish** [comment]
> It's a bit less discoverable that why I think the one tap to add from the row is also good.

Yeah maybe I just have a tendency to try to swipe? I don’t think you should lose the tap option. I just think the swipe could be an easy way too? Just a suggestion 👍

**Comment by Dimillian** [comment]
Yes definitely a good idea :)

**Comment by barclayd** [comment]
This is a great idea. However, from my time implementing a swipeable list using SwiftUI - in order to get a user experience I thought matched what a user would expect, the only way I could get it to work was wrapping a TableViewController in a UIHostingViewController to replace SwiftUI's List. I had tried 'rolling my own' by using a ZStack and adding gestures - the user experience was ok but never quite perfect and so I ultimately dropped it. There's also some workaround for managing multipl edifferent gestures required - which can be seen here on [StackOverflow](https://stackoverflow.com/questions/57700396/adding-a-drag-gesture-in-swiftui-to-a-view-inside-a-scrollview-blocks-the-scroll). Both items are very high on my list for improvements to see in SwiftUI 2.0 😊
